packages/miew/src/ui/ObjectControls.js

```diff
@@ -547,6 +547,12 @@ ObjectControls.prototype.scale = function (factor) {
   if (factor <= 0) {
     return;
   }
+
+  // avoid zoom getting lost in space
+  const tooSmall = this.object.scale.x <= 0.0005 && factor < 1;
+  const tooBig = this.object.scale.x >= 20 && factor > 1;
```

Comment:
Thanks for your contribution and the idea to limit scaling! We appreciate the effort put into this pull request.

Unfortunately, using fixed constant limits for zooming might not be ideal for all scenarios. For instance, zooming out on elements like 2FBU could become problematic, as shown in the image ![image](https://github.com/user-attachments/assets/5fb19ed6-9831-45ef-983c-2cf514b2c4df)). As you can see in the other image ![image](https://github.com/user-attachments/assets/249a741a-5233-432d-bd7b-be22700dbe16), the proposed code restricts zooming out a bit too much in this case.

Let's continue the discussion on alternative approaches in issue #557? Perhaps we can explore a more dynamic scaling solution that adapts to different elements.